packages/apollo-codegen-swift/src/__tests__/language.ts

```diff
@@ -46,6 +66,30 @@ describe("Swift code generation: Basic language constructs", () => {
     `);
   });
 
+  it(`should generate a namespaced fragment`, () => {
+    generator.structDeclaration(
+      {
+        structName: "Hero",
+        adoptedProtocols: ["GraphQLFragment"],
+        namespace: "StarWars"
+      },
+      () => {
+        generator.propertyDeclaration({
+          propertyName: "name",
+          typeName: "String"
+        });
+        generator.propertyDeclaration({ propertyName: "age", typeName: "Int" });
+      }
+    );
+
+    expect(generator.output).toBe(stripIndent`
+      struct Hero: GraphQLFragment {
```

Comment:
I haven't been able to dig super-deep into what's happening here, but is the `adoptedProtocols` thing what shoves this into an extension rather than just into the big namespaced enum?

Reply:
Give me a minute to refresh my memory.

Reply:
@michaelnisi Any recollection?

Reply:
First thing tomorrow morning.

Reply:
No, these changes are just for skipping the `public` modifier if we are in a namespace, see below.

Reply:
@designatednerd Lost interest? 🙂

Reply:
Nope lost the changes in my flood of email 😄 will take another look